### DISCUSSION TITLE
Add Retention time to Docker

### DISCUSSION THREAD

**Discussion by masterwishx** [prometheus/prometheus#10885] [Q&A] [answered]
What is the right way to add --storage.tsdb.retention.time=182d to Docker ?
im using Dcoker in Unraid (Linux x64)
so added in post comands
--config.file=/etc/prometheus/prometheus.yml --storage.tsdb.path=/prometheus --storage.tsdb.retention.time=182d

![image](https://user-images.githubusercontent.com/28630321/174472666-cd2ffee5-29de-4a6b-b113-335fe343cf2d.png)

so now retention = 26w

is it right way to add retention time ?

**Comment by maisi** [comment]
@masterwishx Have you found a solution?

**Reply by masterwishx** [reply, marked as answer]
Yes, this Is the solution
https://github.com/prometheus/prometheus/discussions/10885#discussion-4156775

**Reply by hadjilucasL** [reply]
Just a small note that if you get a permission error when trying to write into the data volume with Unraid, then you might need to set an extra parameter to define the uid/gid of the container user.

`--user 99:100`

In my case this is the uid/gid of the user who owns: `/mnt/user/appdata/prometheus`

`uid=99(nobody) gid=100(users)`

![image](https://github.com/user-attachments/assets/acc09946-79a7-4c1b-bdc2-aa3401a7e335)
